app/src/main/java/in/ac/iitb/gymkhana/iitbapp/fragments/MapFragment.java

```diff
@@ -34,6 +37,20 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
     @Override
     public void onMapReady(GoogleMap googleMap) {
 
+        if (ContextCompat.checkSelfPermission(getActivity(), android.Manifest.permission.ACCESS_FINE_LOCATION) ==
+                PackageManager.PERMISSION_GRANTED &&
+                ContextCompat.checkSelfPermission(getActivity(), android.Manifest.permission.ACCESS_COARSE_LOCATION) ==
+                        PackageManager.PERMISSION_GRANTED){
+            googleMap.setMyLocationEnabled(true);
+            googleMap.getUiSettings().setMyLocationButtonEnabled(true);
+
+        }
+
+
+        LatLngBounds iitbBounds=new LatLngBounds(new LatLng(19.1249000, 72.9046000),new LatLng(19.143522, 72.920000));
+        googleMap.setLatLngBoundsForCameraTarget(iitbBounds);
+        googleMap.setMaxZoomPreference(30);
+        googleMap.setMinZoomPreference(18);
```

Comment:
Download the [Google I/O 2017 app](https://play.google.com/store/apps/details?id=com.google.samples.apps.iosched&hl=en) to know **what** to do. Google hasn't made the source code public yet so you'll have to figure out **how** to do it yourself.